lessons/A-Foundations-and-Tooling/02-spa-project-scaffold.md

```diff
@@ -0,0 +1,113 @@
+
+## 1. Create a New Angular Project
+
+The Angular CLI is the fastest way to get started. By default, it enables strict mode, which is a great starting point for a high-quality application.
+
+- **Action:** Create a new Angular project. The CLI will prompt you for routing and a stylesheet format (we'll use SCSS).
+  ```bash
+  ng new my-awesome-app --style=scss --routing
+  cd my-awesome-app
+  ```
+- **Resource:** [Angular CLI `ng new` documentation](https://angular.io/cli/new)
+
+## 2. Add ESLint and Prettier
+
+A consistent and clean codebase is crucial for maintainability. ESLint helps enforce code quality rules, while Prettier ensures uniform formatting.
+
+- **Action:** Add `angular-eslint` to your project. This schematic will handle the necessary configuration.
+  ```bash
+  ng add @angular-eslint/schematics
+  ```
+- **Action:** Install Prettier and its ESLint integration packages.
+  ```bash
+  npm install --save-dev prettier eslint-config-prettier eslint-plugin-prettier
+  ```
+- **Action:** Create a `.prettierrc.json` file in your project root:
+  ```json
+  {
+    "singleQuote": true,
+    "arrowParens": "avoid",
+    "trailingComma": "all"
+  }
+  ```
+- **Action:** Create a `.prettierignore` file to prevent Prettier from formatting generated files and bundles:
+  ```
+  # Ignore paths for prettier
+  /dist
+  /coverage
+  ```
+- **Resource:** [Best Practices for Prettier & ESLint in Angular](https://www.angulararchitects.io/blog/best-practices-prettier-eslint/)
+
+## 3. Replace Karma/Jasmine with Jest
+
+Jest is a popular, fast, and feature-rich testing framework. We'll replace Angular's default Karma and Jasmine setup with Jest.
+
+- **Action:** Remove Karma and Jasmine packages.
+  ```bash
+  npm uninstall karma karma-chrome-launcher karma-coverage karma-jasmine karma-jasmine-html-reporter jasmine-core jasmine-spec-reporter @types/jasmine
+  ```
+- **Action:** Install Jest and its Angular preset.
+  ```bash
+  npm install --save-dev jest @types/jest jest-preset-angular
+  ```
+- **Action:** Create a `jest.config.js` file in the root of your project:
+  ```javascript
+  module.exports = {
+    preset: 'jest-preset-angular',
+    setupFilesAfterEnv: ['<rootDir>/setup-jest.ts'],
+  };
+  ```
+- **Action:** Create a `setup-jest.ts` file in the root of your project:
+  ```typescript
+  import 'jest-preset-angular/setup-jest';
+  ```
+- **Action:** Update `tsconfig.spec.json` to use Jest types:
+  ```json
+  {
+    "extends": "./tsconfig.json",
+    "compilerOptions": {
+      "outDir": "./out-tsc/spec",
+      "types": ["jest"]
+    },
+    "files": ["src/setup-jest.ts"],
+    "include": ["src/**/*.spec.ts", "src/**/*.d.ts"]
+  }
+  ```
+- **Resource:** [Setting Up Jest in Your Angular Project](https://medium.com/@philip.mutua/setting-up-jest-in-your-angular-16-project-3638ef65f3a3)
+
+## 4. Add Angular Testing Library
+
```

Comment:
Every testing library should have it own lesson with creation of example tests

Reply:
move all testing libraries into separate lesson. Below sections do not provide any value.l